smt/applications/mixed_integer.py

```diff
@@ -43,6 +43,10 @@ def __init__(self, sampling_method_class, design_space, **kwargs):
             category=DeprecationWarning,
         )
         self._design_space = design_space
+        if "random_state" in kwargs:
+            self._design_space.seed = kwargs["random_state"]
```

Comment:
Hmm... AFAIK `seed` is an integer while `random_state` may be an `np.random.RandomState` or an integer. We should check the type here. Maybe the naming seed/random_state is not consistent throughout SMT. We could also start using more type hints.